sim/pokemon.ts

```diff
@@ -1424,10 +1424,11 @@ export class Pokemon {
 			if (this.illusion) {
 				this.ability = ''; // Don't allow Illusion to wear off
 			}
+			const ability = species.abilities[abilitySlot] || species.abilities['0'];
```

Comment:
```suggestion
			const ability = species.abilities[abilitySlot];
```
I don't think the second check is needed here since abilitySlot is set to 0 by default.

Reply:
It's the port from [line 2936 of `data/rulesets.ts`](#diff-c7b144af9efea6f0eb31f259dc22de01d5f5bffa03a19ecaa1b232a8e2584dd3L2936) to handle the case where there's no ability in the old slot.

Reply:
ah, that makes more sense